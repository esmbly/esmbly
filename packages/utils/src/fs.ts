import fs from 'fs';
import util from 'util';

export default {
  readdir: util.promisify(fs.readdir),
  readFile: util.promisify(fs.readFile),
  stat: util.promisify(fs.stat),
  writeFile: util.promisify(fs.writeFile),
};
