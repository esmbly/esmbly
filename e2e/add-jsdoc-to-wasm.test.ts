import fs from 'fs';
import util from 'util';
import path from 'path';
import cp from 'child_process';

jest.setTimeout(10000);

const exec = util.promisify(cp.exec);
const readFile = util.promisify(fs.readFile);
const example = path.resolve(path.join('examples', 'add-jsdoc-to-wasm'));
const dist = path.join(example, 'dist');

describe('E2E: add-jsdoc-to-wasm', () => {
  beforeAll(async () => {
    const { stderr } = await exec(`yarn run esmbly run`, {
      cwd: example,
    });
    if (stderr) {
      // eslint-disable-next-line no-console
      console.log(stderr);
    }
  });

  it('outputs a WebAssembly binary', async () => {
    const result = await readFile(path.join(dist, 'add.wasm'));
    // @ts-ignore
    const { instance } = await WebAssembly.instantiate(result, {});
    expect(instance.exports.add(2, 3)).toEqual(5);
  });

  it('outputs a TypeScript file', async () => {
    const result = await readFile(path.join(dist, 'add.ts'), 'utf8');
    expect(result).toMatchSnapshot();
  });

  it('outputs a AssemblyScript file', async () => {
    const result = await readFile(path.join(dist, 'add.as.ts'), 'utf8');
    expect(result).toMatchSnapshot();
  });
});
