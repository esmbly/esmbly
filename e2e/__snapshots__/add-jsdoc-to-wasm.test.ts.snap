// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E: add-jsdoc-to-wasm outputs a AssemblyScript file 1`] = `
"/**
 * Add two numbers
 * @param {number} a
 * @param {number} b
 * @returns {number}
 */
export function add(a: f64, b: f64): f64 {
  return a + b;
}
"
`;

exports[`E2E: add-jsdoc-to-wasm outputs a TypeScript file 1`] = `
"/**
 * Add two numbers
 * @param {number} a
 * @param {number} b
 * @returns {number}
 */
export function add(a: number, b: number): number {
  return a + b;
}
"
`;

exports[`E2E: add-jsdoc-to-wasm outputs a Wat file 1`] = `
"(module
 (type $FUNCSIG$ddd (func (param f64 f64) (result f64)))
 (type $FUNCSIG$v (func))
 (memory $0 0)
 (table $0 1 funcref)
 (elem (i32.const 0) $null)
 (global $~lib/memory/HEAP_BASE i32 (i32.const 8))
 (export \\"memory\\" (memory $0))
 (export \\"table\\" (table $0))
 (export \\"add\\" (func $add/add))
 (func $add/add (; 0 ;) (type $FUNCSIG$ddd) (param $0 f64) (param $1 f64) (result f64)
  local.get $0
  local.get $1
  f64.add
 )
 (func $null (; 1 ;) (type $FUNCSIG$v)
 )
)
"
`;
